convert2rhel/toolopts.py

```diff
@@ -515,70 +539,118 @@ def _process_cli_options(self):
-    and the logfile
-    """
-    command = " ".join(utils.hide_secrets(sys.argv))
-    loggerinst.info("convert2rhel command used:\n{0}".format(command))
-
-
-def options_from_config_files(cfg_path=None):
+def options_from_config_files(cfg_path):
     """Parse the convert2rhel.ini configuration file.
 
     This function will try to parse the convert2rhel.ini configuration file and
     return a dictionary containing the values found in the file.
 
     .. note::
-       This function will parse the configuration file following a specific
-       order, which is:
-       1) Path provided by the user in cfg_path (Highest priority).
-       2) ~/.convert2rhel.ini (The 2nd highest priority).
-       3) /etc/convert2rhel.ini (The lowest priority).
+        This function will parse the configuration file in the following way:
+
+        1) If the path provided by the user in cfg_path is set (Highest
+           priority), then we use only that.
+
+        Otherwise, if cfg_path is `None`, we proceed to check the following
+        paths:
+
+        2) ~/.convert2rhel.ini (The 2nd highest priority).
+        3) /etc/convert2rhel.ini (The lowest priority).
+
+        In any case, they are parsed in reversed order, meaning that we will
+        start with the lowest priority and go until the highest.
 
     :param cfg_path: Path of a custom configuration file
     :type cfg_path: str
 
     :return: Dict with the supported options alongside their values.
-    :rtype: dict[str, str | None]
+    :rtype: dict[str, str]
     """
-    headers = ["subscription_manager"]  # supported sections in config file
-    # Create dict with all supported options, all of them set to None
-    # needed for avoiding problems with files priority
-    # The name of supported option MUST correspond with the name in ToolOpts()
-    # Otherwise it won't be used
-    supported_opts = {"username": None, "password": None, "activation_key": None, "org": None}
+    # Paths for the configuration files. In case we have cfg_path defined
+    # (meaning that the user entered something through the `-c` option), we
+    # will use only that, as it has a higher priority over the rest
+    config_paths = [cfg_path] if cfg_path else CONFIG_PATHS
+    paths = [os.path.expanduser(path) for path in config_paths if os.path.exists(os.path.expanduser(path))]
+
+    if cfg_path and not paths:
+        raise FileNotFoundError("No such file or directory: %s" % ", ".join(paths))
+
+    found_opts = _parse_options_from_config(paths)
+    return found_opts
 
+
+def _parse_options_from_config(paths):
+    """Parse the options from the given config files.
+
+    .. note::
+        If no configuration file is provided through the command line option
+        (`-c`), we will use the default paths and follow their priority.
+
+    :param paths: List of paths to iterate through and gather the options from
+        them.
+    :type paths: list[str]
```

Comment:
```suggestion
    :return: Return a dict of loaded values under all headers from all the config files with solved priority
    :rtype: dict[str, str]
```
I would add this comment about return value